src/Sylius/Bundle/CoreBundle/Installer/Provider/DatabaseSetupCommandsProvider.php

```diff
@@ -54,16 +67,19 @@ public function getCommands(InputInterface $input, OutputInterface $output, Ques
                 sprintf('The database <info>%s</info> already exists and it has no tables.', $this->getDatabaseName()),
             );
 
-            return $this->getCreateSchemaOrRunMigrationsCommand($outputStyle);
+            return ['doctrine:migrations:migrate' => ['--no-interaction' => true]];
         }
 
-        return $this->getCreateDatabaseWithSchemaCommands($outputStyle);
+        return [
+            'doctrine:database:create' => [],
+            'doctrine:migrations:migrate' => ['--no-interaction' => true],
+        ];
     }
 
     private function isEmptyDatabasePresent(): bool
     {
         try {
-            return 0 === count($this->getSchemaManager()->listTableNames());
+            return 0 === count($this->entityManager->getConnection()->createSchemaManager()->listTableNames());
```

Comment:
Does it makes sense to move
> $this->entityManager->getConnection()->createSchemaManager()

into

> createSchemaManager()

as it was done with `getDatabase()`?

P.S. Is schema manager creation resource/time consuming process? Is it okay to have multiple schema manager created? If not, it can be considered to cache first instance of SchemaManager.